src/client/kotlin/tech/thatgravyboat/skyblockapi/api/profile/sacks/SacksAPI.kt

```diff
@@ -0,0 +1,70 @@
+package tech.thatgravyboat.skyblockapi.api.profile.sacks
+
+import tech.thatgravyboat.skyblockapi.api.data.stored.SacksStorage
+import tech.thatgravyboat.skyblockapi.api.events.base.Subscription
+import tech.thatgravyboat.skyblockapi.api.events.base.predicates.OnlyOnSkyBlock
+import tech.thatgravyboat.skyblockapi.api.events.chat.ChatReceivedEvent
+import tech.thatgravyboat.skyblockapi.api.events.hypixel.ChangedSackItem
+import tech.thatgravyboat.skyblockapi.api.events.hypixel.SacksChangeEvent
+import tech.thatgravyboat.skyblockapi.api.events.screen.ContainerChangeEvent
+import tech.thatgravyboat.skyblockapi.modules.Module
+import tech.thatgravyboat.skyblockapi.utils.extentions.getRawLore
+import tech.thatgravyboat.skyblockapi.utils.extentions.toIntValue
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexGroup
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.anyMatch
+import tech.thatgravyboat.skyblockapi.utils.regex.RegexUtils.match
+import tech.thatgravyboat.skyblockapi.utils.regex.component.ComponentRegex
+import tech.thatgravyboat.skyblockapi.utils.regex.component.match
+import tech.thatgravyboat.skyblockapi.utils.text.TextProperties.stripped
+import tech.thatgravyboat.skyblockapi.utils.text.TextStyle.hover
+import tech.thatgravyboat.skyblockapi.utils.text.TextUtils.splitLines
+
+@Module
+object SacksAPI {
+    // [Sacks] +14 items. (Last 5s.)
+    // [Sacks] -38 items. (Last 5s.)
+    // [Sacks] +38 items, -1 item. (Last 8s.)
+    val sackMessageRegex = ComponentRegex("\\[Sacks\\] (?:(?<gained>\\+[\\d.,]+) items?,?)?\\s*(?:(?<lost>-[\\d.,]+) items?)?\\.\\s*\\(.*")
+    val addedItemsRegex = RegexGroup.CHAT.create("sackapi.changed", " {2}(?<amount>[+-][\\d.,]+) (?<item>.+) \\(.*")
+    val sackTitleRegex = RegexGroup.INVENTORY.create("sackapi.title", ".* Sack")
+    val sackAmountRegex = RegexGroup.INVENTORY.create("sackapi.amount", "Stored: (?<amount>[\\d,.]+)/.*")
+
+    val sackItems: Map<String, Int>
+        get() = SacksStorage.items
+
+    @Subscription
+    @OnlyOnSkyBlock
+    fun onChat(event: ChatReceivedEvent) {
+        sackMessageRegex.match(event.component) {
+            val gainedHoverComponents = it["gained"]?.copy()?.hover?.splitLines() ?: emptyList()
+            val lostHoverComponents = it["lost"]?.copy()?.hover?.splitLines() ?: emptyList()
+            val hoverComponents = gainedHoverComponents + lostHoverComponents
+
+            val changedItems = hoverComponents.mapNotNull {
+                var amount: Int? = null
+                var item: String? = null
+                addedItemsRegex.match(it.stripped, "amount", "item") { (_amount, _item) ->
```

Comment:
use findOrNull and then return the pair inside also remove `.*` from the regex when switching to find